src/libtiled/gidmapper.cpp

```diff
@@ -186,16 +179,96 @@ QByteArray GidMapper::encodeLayerData(const TileLayer &tileLayer,
     return tileData.toBase64();
 }
 
+QByteArray GidMapper::encodeChunkData(const TileLayer &tileLayer,
+                                      int chunkStartX,
+                                      int chunkStartY,
+                                      Map::LayerDataFormat format) const
+{
+    Q_ASSERT(format != Map::XML);
+    Q_ASSERT(format != Map::CSV);
+
+    int startX = chunkStartX;
+    int startY = chunkStartY;
+    int endX = chunkStartX + CHUNK_SIZE - 1;
+    int endY = chunkStartY + CHUNK_SIZE - 1;
+
+
+    QByteArray chunkData;
+    chunkData.reserve((endX - startX + 1) * (endY - startY + 1) * 4);
+
+    for (int y = startY; y <= endY; ++y) {
+        for (int x = startX; x <= endX; ++x) {
+            const unsigned gid = cellToGid(tileLayer.cellAt(x, y));
+            chunkData.append((char) (gid));
+            chunkData.append((char) (gid >> 8));
+            chunkData.append((char) (gid >> 16));
+            chunkData.append((char) (gid >> 24));
+        }
+    }
+
+    if (format == Map::Base64Gzip)
+        chunkData = compress(chunkData, Gzip);
+    else if (format == Map::Base64Zlib)
+        chunkData = compress(chunkData, Zlib);
+
+    return chunkData.toBase64();
+}
+
 GidMapper::DecodeError GidMapper::decodeLayerData(TileLayer &tileLayer,
                                                   const QByteArray &layerData,
+                                                  Map::LayerDataFormat format) const
```

Comment:
Similarly, this function could take a `QRect` to specify the target area of the decoded layer data.